### ISSUE TITLE
If "Firefox for Android" was checked for a previous addon version it should remain checked on submitting a new version

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-server#10398]
From: https://github.com/mozilla/addons-frontend/issues/7394

---

### Describe the problem and steps to reproduce it:
1. Submit a new addon to AMO.
2. Mark ` Firefox for Android` under `Which applications is this version compatible with?`
3. Submit an updated version of the same addon to AMO.
4. Check what the default state is of the `Firefox for Android` checkbox

### What happened?
The checkbox was not checked.

### What did you expect to happen?
The checkbox should be checked as previous version were also compatible with Firefox for Android. Now it is incredibly easy to miss that checkbox when submitting your addon and accidentally withhold updates from a user group. 

### Anything else we should know?
This issue is a result of the discussion in [this bugzilla issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1519725)

**Comment by buttercookie42** [comment]
And if I remember correctly, subsequently you also have to re-add Android in the compatibility data for that version as well, even if you do remember to check the checkbox on the initial update upload.

**Comment by jvillalobos** [comment]
We'll probably ramp up Android support in the coming months, and we might add Recommended extension Android support very soon, so we should fix this if still valid.

**Comment by ioanarusiczki** [comment]
@willdurand I verified this on AMO dev with FF69(Win10): 

- when submitting an add-on "On this site", " Firefox for Android" will stay checked when the next version is submitted. 

![on this site works](https://user-images.githubusercontent.com/33448286/67403024-0948d100-f5ba-11e9-9876-52501e298f14.gif)

- when submitting an add-on "On your own", "Firefox for Android" is not remembered for the next submission of an updated version. 

![on your own](https://user-images.githubusercontent.com/33448286/67403273-588f0180-f5ba-11e9-9276-97d1c08c17f7.gif)

I don't know if the second scenario works as expected.

**Comment by EnTeQuAk** [comment]
That is correct, the query doesn't take unlisted versions into account but only looks into the most recent publicly listed version.

Somehow that felt straightforward to me, especially since unlisted versions aren't really meant to be a "follow-up" to a publicly listed version.

But maybe I'm wrong though, any thoughts on that - maybe @jvillalobos?

**Comment by jvillalobos** [comment]
That setting is only useful for public versions, so I think that makes sense. Now I'm thinking we shouldn't show it for unlisted versions, but that's a separate issue.

**Comment by EnTeQuAk** [comment]
Perfect, thanks for chiming in!
